include/snowflake/Simba_CRTFunctionSafe.h

```diff
@@ -0,0 +1,337 @@
+    /// @param in_argPtr        Pointer to list of arguments.
+    /// 
+    /// @return The number of bytes written to the buffer, not counting the terminating null 
+    /// character; -1 if the truncation or error occurred.
+    inline int sb_vsnprintf(
+        char* out_buffer,
+        size_t in_sizeOfBuffer,
+        size_t in_sizeToWrite,
+        const char* in_format,
+        va_list in_argPtr)
+    {
+        if (in_sizeToWrite >= in_sizeOfBuffer) return -1;
+
+        return _vsnprintf_s(out_buffer, in_sizeOfBuffer, _TRUNCATE, in_format, in_argPtr);
+    }
+
+    /// @brief Write formatted data to a string. 
+    /// 
+    /// @param out_buffer       Storage location for output. (NOT OWN)
+    /// @param in_sizeOfBuffer  Size of buffer in characters. 
+    /// @param in_format        Format control string. (NOT OWN)
+    /// @param ...              Optional arguments for printf style conversions.
+    /// 
+    /// @return The number of bytes written to the buffer, not counting the terminating null 
+    /// character; -1 if an error occurred.
+    inline int sb_sprintf(
+        char* out_buffer,
+        size_t in_sizeOfBuffer,
+        const char* in_format,
+        ...)
+    {
+        if (!in_sizeOfBuffer) return -1;
+
+        va_list formatParams;
+        va_start(formatParams, in_format);
+        int bytesWritten = sb_vsnprintf(out_buffer, in_sizeOfBuffer, in_sizeOfBuffer - 1, in_format, formatParams);
+        va_end(formatParams);
+
+        return bytesWritten;
+    }
+
+    /// @brief Reads formatted data from a string. 
+    /// 
+    /// Note: A buffer size parameter is required when you use the type field characters 
+    /// c, C, s, S, or string control sets that are enclosed in []. 
+    ///
+    /// @param in_buffer        Stored data. (NOT OWN)
+    /// @param in_format        Format control string. (NOT OWN)
+    /// @param ...              Optional arguments.
+    /// 
+    /// @return The number of fields that are successfully converted and assigned;
+    /// -1 if an error occurred.
+    inline int sb_sscanf(
+        const char* in_buffer,
+        const char* in_format,
+        ...)
+    {
+        va_list formatParams;
+        va_start(formatParams, in_format);
+        int ret = vsscanf_s(in_buffer, in_format, formatParams);
+        va_end(formatParams);
+
+        return ret;
+    }
+
+#else   // For all other platforms except Windows
+
+    /// @brief Copy a string.
+    /// 
+    /// @param out_dest         Destination string buffer. (NOT OWN)
+    /// @param in_destSize      Size of the destination string buffer.
+    /// @param in_src           Null-terminated source string buffer. (NOT OWN)
+    /// 
+    /// @return The destination string; NULL if an error occurred. (NOT OWN)
+    inline char* sb_strcpy(
+        char* out_dest,
+        size_t in_destSize,
+        const char* in_src)
+    {
+        return strcpy(out_dest, in_src);
```

Comment:
oh, wouldn't this really check the destination buffer size?

Reply:
This is the code for non-windows and copied from SimbaEnginSDK/Include/Support/CRTFunctionSafe.h
Those safe CRT functions are made to meet Microsoft SDL requirement so basically Windows only.